client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java

```diff
@@ -79,6 +91,36 @@ public class SparkShuffleManager implements ShuffleManager {
   private long sendBufferPoolCheckInterval;
   private long sendBufferPoolExpireTimeout;
 
+  // track appShuffleId -> shuffleId Set in executor for cleanup
+  static class AppShuffleIdTracker {
+    private ConcurrentHashMap<Integer, HashSet<Integer>> shuffleIdMap =
+        JavaUtils.newConcurrentHashMap();
+
+    public void track(int appShuffleId, int shuffleId) {
+      HashSet<Integer> shuffleIds =
+          shuffleIdMap.computeIfAbsent(
+              appShuffleId,
+              new Function<Integer, HashSet<Integer>>() {
+                @Override
+                public HashSet<Integer> apply(Integer id) {
+                  return new HashSet<Integer>(shuffleId);
```

Comment:
The constructor is specifying initial capacity.

```suggestion
                  return new HashSet<Integer>();
```

Reply:
Right. I guess the intention is to put `shuffleId` when initialize the set, but since L110 always adds into the set, we don't need to add here.

Reply:
an unfortunate mistake...
I've fixed it, thanks!